avocado/core/test.py

```diff
@@ -312,21 +305,28 @@ def _get_tagged_name(self, logdir):
 
         :param logdir: Log directory being in use for result storage.
 
-        :return: Unique test name
+        :return: Unique test name and the logdir
         """
         name = self.name
         if self.tag is not None:
             name += ".%s" % self.tag
         tag = 0
         tagged_name = name
-        safe_tagged_name = astring.string_to_safe_path(tagged_name)
-        while os.path.isdir(os.path.join(logdir, safe_tagged_name)):
+        # The maximal length on ext4+python2.7 is 255 chars.
+        safe_tagged_name = astring.string_to_safe_path(tagged_name[:240])
```

Comment:
I didn't get the 240 limit here. If the tag is limited to 9999, so the worst case is `tagged_name[:251]` to fit the 255 limit.